### PR TITLE
awsecscontainermetrics: check for empty network rate stats and set zero

### PR DESCRIPTION
**Pull request by hossain-rayhan**
**Description:** <Describe what has changed.>
Fixing a bug: `awsecscontainermetrics` receiver works fine for the latest version of AWS ECS Agent. However, if customer uses an older version than 1.44.0, it throws a null pointer exception. This change adds a check for empty network stats (which occurs only if the Agent is under 1.44.0) and sets zero.

**Link to tracking Issue:** 
#1259 

**Testing:** 
Unit test added

### CODE REVIEW

**Comment by hossain-rayhan** [comment]
Hi @tigrannajaryan , could you have a look here. It's a two line changes to fix a known bug.

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func getContainerMetrics(stats ContainerStats) ECSMetrics {
 	netStatArray := getNetworkStats(stats.Network)
 
 	storageReadBytes, storageWriteBytes := extractStorageUsage(&stats.Disk)
+	floatZero := float64(1.0)
```
Comment (mxiamxia) [inline receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go:29]: The variable name is `Zero` but why it has value 1? 
And why the metric value is 1 when the datasource is absent?

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go [hunk 2]
```diff
@@ -42,8 +43,13 @@ func getContainerMetrics(stats ContainerStats) ECSMetrics {
...
 
-	m.NetworkRateRxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
-	m.NetworkRateTxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+	if stats.NetworkRate == (NetworkRateStats{}) {
+		m.NetworkRateRxBytesPerSecond = floatZero
+		m.NetworkRateTxBytesPerSecond = floatZero
+	} else {
+		m.NetworkRateRxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+		m.NetworkRateTxBytesPerSecond = *stats.NetworkRate.TxBytesPerSecond
+	}
```
Comment (mxiamxia) [inline receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go:53]: Can I learn where did the NPE throw? `*stats.NetworkRate.TxBytesPerSecond` shouldn't give NPE when `*stats.NetworkRate` is empty structure.

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go (hunk 1 above)
Comment (hossain-rayhan) [reply receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go:29]: good catch. It should be zero.

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go (hunk 2 above)
Comment (hossain-rayhan) [reply receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/metrics_helper.go]: It's in the same line you mentioned. To me, when we have an empty structure, we don't have references for `RxBytesPerSecond` and `TxBytesPerSecond`. When we are dereferencing values from these pointers, it's throwing a null pointer exception. At least, thats my understanding.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**